pack.go

```diff
@@ -90,7 +119,60 @@ func Pack(ctx context.Context, pusher content.Pusher, layers []ocispec.Descripto
 		Size:      int64(len(manifestBytes)),
 	}
 
-	// push manifest.
+	// push manifest
+	if err := pusher.Push(ctx, manifestDesc, bytes.NewReader(manifestBytes)); err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
+		return ocispec.Descriptor{}, fmt.Errorf("failed to push manifest: %w", err)
+	}
+
+	return manifestDesc, nil
+}
+
+// PackArtifact packs the given blobs, generates an ORAS Artifact Manifest for
+// the pack, and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
+// Returns ErrMissingArtifactType if artifactType is empty.
+// Reference: https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md
+func PackArtifact(ctx context.Context, pusher content.Pusher, blobs []artifactspec.Descriptor, artifactType string, opts PackArtifactOptions) (ocispec.Descriptor, error) {
+	if artifactType == "" {
+		// artifactType is required for ORAS Artifact Manifest
+		return ocispec.Descriptor{}, ErrMissingArtifactType
+	}
+
+	if opts.ManifestAnnotations == nil {
+		opts.ManifestAnnotations = make(map[string]string)
+	}
+
+	_, ok := opts.ManifestAnnotations[AnnotationArtifactCreated]
+	if !ok {
+		// set creation time in RFC3339 format, if not provided
+		// reference: https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties
+		now := time.Now().UTC()
+		opts.ManifestAnnotations[AnnotationArtifactCreated] = now.Format(time.RFC3339)
```

Comment:
Map is a reference type. Setting `opts.ManifestAnnotations[AnnotationArtifactCreated] = now.Format(time.RFC3339)` may modify the original map, rendering `PackArtifact()` not concurrent safe.